app/src/main/java/ch/hikemate/app/utils/MapUtils.kt

```diff
@@ -186,13 +206,48 @@ object MapUtils {
             MapScreen.USER_LOCATION_MARKER_ICON_SIZE,
             MapScreen.USER_LOCATION_MARKER_ICON_SIZE,
             Bitmap.Config.ARGB_8888)
-    val canvas = android.graphics.Canvas(bitmap)
+    val canvas = Canvas(bitmap)
     originalDrawable?.setBounds(0, 0, canvas.width, canvas.height)
     originalDrawable?.draw(canvas)
 
     return BitmapDrawable(context.resources, bitmap)
   }
 
+  /**
+   * Creates a circular icon with a fill color and a stroke color. The icon is used to represent the
+   * starting point of a hike on the map.
+   *
+   * @param context The context where the icon will be used
+   * @param fillColor The color to fill the circle
+   * @return The circular icon with the specified fill and stroke colors
+   */
+  private fun createCircularIcon(context: Context, fillColor: Int): BitmapDrawable {
+    // Create a mutable bitmap
+    val bitmap =
+        Bitmap.createBitmap(
+            MapScreen.HIKE_STARTING_MARKER_ICON_SIZE,
+            MapScreen.USER_LOCATION_MARKER_ICON_SIZE,
+            Bitmap.Config.ARGB_8888)
+    val canvas = Canvas(bitmap)
```

Comment:
Question: I don't understand what do this do? Why do you use a **_USER_LOCATION_**_MARKER_ICON_SIZE?

Reply:
Good catch! my bad